src/PrestaShopBundle/Resources/config/services/bundle/new_api.yml

```diff
@@ -0,0 +1,64 @@
+services:
+  _defaults:
+    public: true
+
+  prestashop.api.repository.client:
+    class: PrestaShopBundle\Security\OAuth2\Repository\ClientRepository
+    arguments:
+      - '@security.user.provider.concrete.oauth2'
+      - '@security.user_password_encoder.generic'
+
+  prestashop.api.repository.access_token:
+    class: PrestaShopBundle\Security\OAuth2\Repository\AccessTokenRepository
+
+  prestashop.api.repository.scope:
+    class: PrestaShopBundle\Security\OAuth2\Repository\NullScopeRepository
+
+  prestashop.api.grant.client_credentials:
+    class: League\OAuth2\Server\Grant\ClientCredentialsGrant
+
+  prestashop.api.private_key:
+    class: League\OAuth2\Server\CryptKey
+    arguments: [ '%api_private_key%' ]
```

Comment:
I think these parameters should come from bundle configuration and not depend on project parameters

Reply:
I kinda agree but there's no other mechanism that allows us to create config in the bundle during install. Or maybe the private/public key should go on the project rather than on the bundle?

Reply:
So do we plan on having some configuration page for this? It's only necessary when we are using the Core authentication server right? Or is it common to any implementation including and external one from a module?

Reply:
@jolelievre no actually those keys are generated during the installation, like the cookie_key.